### ISSUE TITLE
Add Fedora image

### ISSUE THREAD

**Issue by kolyshkin** [actions/runner-images#10802]
### Tool name

Fedora

### Tool license

mostly GPL

### Add or update?

- [X] Add
- [ ] Update

### Desired version

40

### Brief description of tool

Please consider having more VM images. In particular, it would be awesome to have Fedora Linux.

I am asking for Fedora specifically because it is a distro which has unique distinct features not available elsewhere. From my recent experience, this is 
*  **SELinux**. Ubuntu comes with Apparmor by default, and while it is possible to enable SELinux, it requires a reboot which (to my best understanding) is not possible in GHA. In particular, I am a maintainer of https://github.com/opencontainers/selinux Go package, which can't be tested in GHA because, naturally, it needs SELinux for most of its tests.

The wonderful world of Linux doesn't start or end with Ubuntu, and Fedora, in particular, is useful for CI. It is actively developed, has lots of users, backed up by a reputable company (disclaimer: I work for Red Hat, but not on Fedora), and a few things about it are distinctive enough.

NOTE that Fedora provides and maintains cloud images for Azure (which, I believe, is used by GHA) here: https://fedoraproject.org/cloud/download. I believe that just using this image will be sufficient for many use cases.

I am well aware that I can run fedora container, but some features (like SELinux above) can't be used unless supported on the host, and SELinux is one of them. I am pretty sure there are others, I just came across SELinux. Some other requests about having Fedora list other reasons, so let me just list those:

* #1443
* #45
* #898
* #2307

The last one is the one I filed almost 4 years ago. It got 40+ likes and a lively discussion. For runc, we ended up using Cirrus CI.

----

I am not trying to pry or annoy the github team. In https://github.com/actions/virtual-environments/issues/1443#issuecomment-684790353 @maxim-lobanov says:

> Since we see popularity of this request, we will keep it on mind and will be able to revisit this decision in future.

So, maybe the time to revisit is? :+1: 😉 

### URL for tool's homepage

https://fedoraproject.org/cloud/download

### Platforms where you need the tool

- [ ] Azure DevOps
- [X] GitHub Actions

### Runner images where you need the tool

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] Ubuntu 24.04
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [ ] macOS 15
- [ ] macOS 15 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Can this tool be installed during the build?

No.

Fedora can be run in a container, but if we need features related to Fedora kernel or systemd, those are impossible to have without running Fedora natively (i.e. not in a container).

### Are you willing to submit a PR?

yes

**Comment by Dylan-M** [comment]
I seconded #2307, and my needs have not changed since.

Additionally, I would agree with another comment on that previous issue: Alpine would be good to have as well.

I do not have the infrastructure to maintain self hosted runners, and doing the work inside of a container is not suitable.

**Comment by lakshminarayana02** [comment]
Dear @kolyshkin 

We acknowledge the need for a Fedora image to be accessible on GitHuh Actions  for executing workflows. Fedora may offer unique capabilities that are not present in the current Ubuntu image  like RPM-based package management, cgroupv2, or new syscalls  etc. 

We appreciate your patience while  we validate your request which might require a bit of time for a response

**Comment by Kichura** [comment]
Will this type of image make use of Fedora IoT, CoreOS or something else in this case?

**Comment by lakshminarayana02** [comment]
Hi
Following discussions with the appropriate stakeholders, we learned that there are no plans to add  further linux images for  GHA.

**Comment by lakshminarayana02** [comment]
Hi
Following discussions with the appropriate stakeholders, we learned that there are no plans to add further linux images for GHA.

**Comment by Dylan-M** [comment]
> Hi Following discussions with the appropriate stakeholders, we learned that there are no plans to add further linux images for GHA.

Which is exactly the point. Ya'll keep telling us that, and we keep opening these issues to communicate that we, the community that uses the product, want and need more. There is no quicker way for a company to drive users to their competitors than to keep ignoring common, high demand, and reasonable asks. This issue has been raised a half dozen times with overwhelmingly positive community response. The complete disregard from the GitHub staff is, frankly, both disappointing and terrifying.

**Comment by Kichura** [comment]
> Hi Following discussions with the appropriate stakeholders, we learned that there are no plans to add further linux images for GHA.

I hereby fail to see the reasoning why this is the conclusion behind this kind of closure, while maintaining two linux distributions is considered painful in nature - having fedora is really not a concern when it comes to maintenance management as most people do have the ability to submit pull requests to help resolve problems in the event there is one at some point.

And because of this, this makes me now assume all fedora CLI end-users will now just end up using seperate workflow programs instead to run their tests instead of having to relay on actions directly which is the unfortunate context of it.

Basically the tl;dr: This is informative and unfortunate.

**Comment by Kichura** [comment]
Moved to [this issue](https://github.com/actions/runner-images/issues/14075) to express my interest further, However i changed the subject to AlmaLinux instead as Fedora gets updated too frequently.

**Comment by Dylan-M** [comment]
> Moved to [this issue](https://github.com/actions/runner-images/issues/14075) to express my interest further, However i changed the subject to AlmaLinux instead as Fedora gets updated too frequently.

Honestly, with how little GitHub cares about meeting the needs of their users these days, we've moved our runners off to a 3rd party platform that is more reliable, and offers more options. As well as caring about our needs.

**Comment by HinTak** [comment]
I too am needing this as I am using the ubuntu runner purely to run a fedora container to build rpms.

**Comment by mcfriend99** [comment]
@HinTak how do you currently handle this? I'm also looking for a way to build rpm packages via fedora

**Comment by HinTak** [comment]
> [@HinTak](https://github.com/HinTak) how do you currently handle this? I'm also looking for a way to build rpm packages via fedora

See https://github.com/skia-python/skia-python/blob/03e3b2aed824fc4e888e4e6fd6fcdb1b19543f89/.github/workflows/ci.yml#L163

**Comment by hassaanirshad** [comment]
Thank you for providing the existing runner images. They are extremely helpful. And as other commenters requested, a Fedora image would be really helpful as well. Thank you.
